### PR TITLE
Added matmul method to the Paddle frontend

### PR DESCRIPTION
**Pull request by he11owthere**
closes https://github.com/unifyai/ivy/issues/22596

image

### CODE REVIEW

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 1]
```diff
@@ -100,6 +100,11 @@ def dim(self):
     def abs(self):
         return paddle_frontend.abs(self)
```
Comment (aparajith21) [inline ivy/functional/frontends/paddle/tensor/tensor.py:102]: Please add this here: `@with_unsupported_dtypes({"2.5.1 and below": ("float16", "bfloat16")}, "paddle")`

**Comment by aparajith21** [comment]
Hi, could you please comment add_frontend_checklist, go over the items there, and check if they're all good and go over any test failures locally. Please request review after that and I will check or let me know if there's any help needed meanwhile!

**Comment by he11owthere** [comment]
add_frontend_checklist

**Comment by aparajith21** [comment]
Tests seem to be failing because of dtype errors, you may want to look into the same on the test side and also clean up the code a bit so it is all in one place in the file! Thanks

**Pull request merged**